File: 1_programlamaya_giris/bolum_sonu_sorular/1.8_Girilen_Sayinin_50_den_buyuk_kucuk_veya_esitligini_gosteren_program.cs
using System.Diagnostics;
using System;

namespace programlama_algoritma_giris
{
    internal class Program
    {
        static void Main(string[] args)
        {

            try // try-catch bloğu ile kullanıcıdan alınan değerin float olup olmadığı kontrol ediliyor.
            {
                Console.Write("Lütfen bir değer girin: "); // Kullanıcıdan bir değer isteniyor.
                string userInput = Console.ReadLine(); // Girilen değer string olarak alınıyor.

                float floatValue = float.Parse(userInput); // Girilen değer float olarak değişkene atanıyor.

                if(floatValue < 50) // Girilen değer 50'den küçükse ekrana "Girdiğiniz Sayı 50'den Küçük." yazdırılıyor.
                {
                    Console.WriteLine("Girdiğiniz Sayı 50'den Küçük.");
                }
                else if (floatValue > 50) // Girilen değer 50'den büyükse ekrana "Girdiğiniz Sayı 50'den Büyük." yazdırılıyor.
                {
                    Console.WriteLine("Girdiğiniz Sayı 50'den Büyük.");
                } else // Girilen değer 50'ye eşitse ekrana "Girdiğiniz Sayı 50'ye Eşit." yazdırılıyor.
                {
                    Console.WriteLine("Girdiğiniz sayı 50'ye eşit.");
                }
            }
            catch (FormatException) // Girilen değer float değilse ekrana "Hatalı giriş! Bir float değer girin." yazdırılıyor.
            {
                Console.WriteLine("Hatalı giriş! Bir float değer girin."); // Hata mesajı.
            }
        }
    }
}

File: 1_programlamaya_giris/bolum_sonu_sorular/1.8_Girilen_Sayinin_50_den_buyuk_kucuk_veya_esitligini_gosteren_program.py
value = input("Lütfen bir değer girin: ") #Kullanıcıdan bir değer isteniyor.
try:
    value = float(value) #Kullanıcının girdiği değer float olarak değiştiriliyor.

    if value > 50: #Girilen değer 50'den büyükse ekrana "Girilen değer 50'den büyük." yazdırılıyor.
        print("Girilen değer 50'den büyük.")
    elif value < 50: #Girilen değer 50'den küçükse ekrana "Girilen değer 50'den küçük." yazdırılıyor.
        print("Girilen değer 50'den küçük.")
    else: #Girilen değer 50'ye eşitse ekrana "Girilen değer 50'ye eşit." yazdırılıyor.
        print("Girilen değer 50'ye eşit.")

except: #Girilen değer float değil ise ekrana "Hatalı giriş! Lütfen bir sayı girin." yazdırılıyor.
    raise TypeError('Hatalı giriş! Lütfen bir sayı girin.')

